sdk/tables/azure-tables/src/serialization.ts

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import { base64Encode, base64Decode } from "./utils/bufferSerializer";
+
+const Edm = {
+  Binary: "Edm.Binary",
+  Boolean: "Edm.Boolean",
+  DateTime: "Edm.DateTime",
+  Double: "Edm.Double",
+  Guid: "Edm.Guid",
+  Int32: "Edm.Int32",
+  Int64: "Edm.Int64",
+  String: "Edm.String"
+};
+
+type supportedTypes = boolean | string | number | Date | Uint8Array;
+
+type serializedType = {
+  value: supportedTypes;
+  type?: string;
+};
+
+function serializePrimitive(value: any): serializedType {
+  const serializedValue: serializedType = { value };
+  if (typeof value === "boolean" || typeof value === "string" || typeof value === "number") {
+    serializedValue.value = value;
+  } else if (value instanceof Date) {
+    serializedValue.value = value;
+    serializedValue.type = Edm.DateTime;
+  } else if (value instanceof Uint8Array) {
+    serializedValue.value = base64Encode(value);
+    serializedValue.type = Edm.Binary;
+  } else {
+    throw new Error(`Unknown EDM type ${typeof value}`);
+  }
+
+  return serializedValue;
+}
+
+function serializeObject(obj: { value: any; type: string }): serializedType {
```

Comment:
would it be worth providing stronger typing for `type` since it has to be one of a set of string literals we support?